src/components/LoginForm/LoginForm.js

```diff
@@ -28,31 +26,26 @@ export class LoginForm extends Component {
       username,
       password
     };
-    this.setState({ error: '', loading: true });
     
     e.preventDefault();
     this.signIn(user);
   }
 
   signIn = async (user) => {
     const { addUser } = this.props;
-
-    try {
-      const result = await requests.signIn(user);
+    const result = await requests.signIn(user);
+    if (result) {
       const { user: validUser, token } = result.signIn;
-      if (token && validUser) {
-        localStorage.setItem('token', JSON.stringify(token));
-        addUser(validUser);
-        this.setState({ success: true, loading: false });
-      }
-    } catch (error) {
-      const { message } = error.graphQLErrors[0];
-      this.setState({ error: message, loading: false });
-    }    
+      localStorage.setItem('token', JSON.stringify(token));
+      addUser(validUser);
+      this.setState({ success: true });
+    }
```

Comment:
this is where the conversion to redux isLoading/error begins, the try catch is no longer needed as it became part of the primary fetch function